### PR TITLE
Merge fsleyes-plugin-shimming-toolbox repo in Shimming Toolbox

### PR DESCRIPTION
**Pull request by po09i**
## Description
This PR merges the [fsleyes-plugin-shimming-toolbox repository](https://github.com/shimming-toolbox/fsleyes-plugin-shimming-toolbox) as a subdirectory in Shimming Toolbox

Todo:
- [x] Merge the plugin repo while keeping the git history
- [x] Update the installation script for MacOS and Linux
- [x] Add a PLUGIN option that defaults to true if we want to not install the plugin
- [x] Update windows install script
- [x] The CI has been updated to run both ST and the plugin tests
- [x] Deleted old Shimming Toolbox API examples not necessary
- [x] Move config files in the chiming-toolbox package directory
- [x] Update installation documentation to direct to ST website
- [x] Update dev installation instructions

Note:
This might need to not be a squash and merge. 
Update: This should be a standard merge

### Info about the merge
Taken form a response [here](https://stackoverflow.com/questions/6426247/merge-git-repository-in-subdirectory)
```
git clone git@github.com:shimming-toolbox/shimming-toolbox.git
cd shimming-toolbox
git checkout ad/merge-plugin
cd ..
git clone git@github.com:shimming-toolbox/fsleyes-plugin-shimming-toolbox.git
cd fsleyes-plugin-shimming-toolbox
mkdir fsleyes-plugin-shimming-toolbox
# Move all files of the repo into fsleyes-plugin-shimming-toolbox and git add all files so that they are staged for commit as a rename
git commit -m "Move to subdirectory"
cd ../shimming-toolbox
git remote add -f fsleyes-plugin-shimming-toolbox ../fsleyes-plugin-shimming-toolbox
git merge -s ours --no-commit --allow-unrelated-histories fsleyes-plugin-shimming-toolbox/main 
git read-tree --prefix= -u fsleyes-plugin-shimming-toolbox/main
git commit -m "Subtree merge plugin into fsleyes-plugin-shimming-toolbox folder"
git log fsleyes-plugin-shimming-toolbox --follow
```

## Linked issues
Related to #528

### CODE REVIEW

**Comment by mguaypaq** [comment]
@po09i good job with the repository merge! This looks pretty clean. I like that you have separate commits for:
1. moving the files in each repository to a subfolder,
2. doing a plain merge with no modifications (cd6f5a216cec74a9f690421e4fdac79fbedb00f5), and
3. doing the necessary cleanup and deduplication afterwards.

That should make it a lot easier to investigate in the future if there are questions about the merge.

I agree with this note:
> This might need to **not** be a squash and merge

**Comment by 4rnaudB** [comment]
When I check out to ad/merge-plugin, I get an extra shimmingtoolbox/shimminftoolbox folder and a shimmingtoolbox/test/ folder filled with .pyc files

**Comment by po09i** [comment]
> When I check out to ad/merge-plugin, I get an extra shimmingtoolbox/shimminftoolbox folder and a shimmingtoolbox/test/ folder filled with .pyc files

This seems to be local python files. What is tracked by GitHub seems to be good.

**Comment by 4rnaudB** [comment]
Confirmed. When changing to this branch you need to `make install CLEAN=true` to make sure everything is reset.

**Comment by 4rnaudB** [comment]
There is a bug where the slices are displayed in the terminal like this:
(np.int64(0),), (np.int64(1),), (np.int64(2),), (np.int64(3),), (np.int64(4),), (np.int64(5),), (np.int64(6),), (np.int64(7),), (np.int64(8),), (np.int64(9),), (np.int64(10),), (np.int64(11),), (np.int64(12),), (np.int64(13),), (np.int64(14),), (np.int64(15),), (np.int64(16),), (np.int64(17),), (np.int64(18),), (np.int64(19),)]

instead of 
[(0, ), (1, ), ...]

**Comment by po09i** [comment]
> There is a bug where the slices are displayed in the terminal like this:
(np.int64(0),), (np.int64(1),), (np.int64(2),), (np.int64(3),), (np.int64(4),), (np.int64(5),), (np.int64(6),), (np.int64(7),), (np.int64(8),), (np.int64(9),), (np.int64(10),), (np.int64(11),), (np.int64(12),), (np.int64(13),), (np.int64(14),), (np.int64(15),), (np.int64(16),), (np.int64(17),), (np.int64(18),), (np.int64(19),)]

The problem comes from numpy 2.0

**Comment by po09i** [comment]
@4rnaudB the issue you mentioned should be fixed now ([686c3fd](https://github.com/shimming-toolbox/shimming-toolbox/pull/533/commits/686c3fdede3ab48fa450feeb2c983a29fd81b3de))

**Comment by 4rnaudB** [comment]
Indeed, did you find out why this PR changed this line?

**Review by 4rnaudB — APPROVED** [review]

**Comment by 4rnaudB** [comment]
Also, I would make sure everyone deletes and reinstalls shimming-toolbox after this PR/release because of the problems I faced with double folders and stuff.

File: docs/source/user_section/installation/install_on_mac_and_linux.rst [hunk 1]
```diff
@@ -27,19 +27,24 @@ Installation Procedure for macOS and Linux
     They will not interfere if you already have ``FSLeyes`` or ``dcm2niix`` installed. If you
     have previously installed ``FSLeyes``, that version will not support the GUI.
 
-Open a Terminal and run the following commands.
+Open a Terminal, navigate where you want to download the source code and run the following commands.
 
-First, download the FSLeyes plugin:
+.. Note::
```
Comment (jcohenadad) [inline docs/source/user_section/installation/install_on_mac_and_linux.rst]:
```suggestion
.. Warning::
```

**Comment by jcohenadad** [comment]
Not needed anymore:

https://github.com/shimming-toolbox/shimming-toolbox/blob/66179cec87c2e382e92b11c3150f1f83dd09250e/docs/source/user_section/installation/install_on_mac_and_linux.rst?plain=1#L66

**Comment by jcohenadad** [comment]
Suggest making it CLEAN=true to avoid possible issues:

https://github.com/shimming-toolbox/shimming-toolbox/blob/66179cec87c2e382e92b11c3150f1f83dd09250e/Makefile#L4

**Comment by jcohenadad** [comment]
suggestion: 

https://github.com/shimming-toolbox/shimming-toolbox/blob/66179cec87c2e382e92b11c3150f1f83dd09250e/docs/source/user_section/installation/install_on_mac_and_linux.rst?plain=1#L60

replace with:
~~~
The ``shimming-toolbox`` command launches FSLeyes with GUI support. It takes a few seconds to open.
~~~

**Review by jcohenadad — COMMENTED** [review]
Tested, works well 😊  added minor comments

**Pull request merged**